### ISSUE TITLE
Incorrect path detection

### ISSUE THREAD

**Issue by eugeniumegherea** [krasimir/navigo#191]
Hello. 
Today found some ugly bug.
Sample: 
`router.on({
					'/search/:v/:q': {
						as: 'search',
						uses: function () {
							console.log('search', arguments[0]);
						}
					},
					'/locations': {
						as: 'home',
						uses: function () {
							console.log('root');
						}
					}
				});	`

The route search will never work if in its parametrized url will be string '/locations', which is another route path. The callback is never called
In other words, navigation to `http://127.0.0.1:8080/#/search/locations/d` does not trigger callback, however, `http://127.0.0.1:8080/#/search/notlocations/d` will work

**Comment by buff32** [comment]
I'm interested in working on a PR for this, but I've not been able to replicate it in either Chrome or Firefox.

This is the test page I've used:

```
<!DOCTYPE html>
<html>
  <head>
    <title>Navigo test</title>
    <script src="https://cdn.jsdelivr.net/npm/navigo@7.1.2/lib/navigo.min.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
    <script>
    $(function() {
      var router = new Navigo(null, true, '#');

      router.on(
        {
          '/search/:v/:q': {
            as: 'search',
            uses: function () {
              console.log('search', arguments[0]);
            }
          },
          '/locations': {
            as: 'home',
            uses: function () {
              console.log('root');
            }
          }
        }
      );

      router.resolve();
    });
    </script>
  </head>
  <body></body>
</html>
```

Console outputs accessing "#/search/notlocations/d":
`search {v: "notlocations", q: "d"}`

Similarly, for "#/search/locations/d":
`search {v: "locations", q: "d"}`

Could you provide some sample code that demonstrates the problem please?

**Comment by krasimir** [comment]
There is a new version 8.0.0 of the library and the bug can't be reproduced there. Navigo got re-written so probably the new implementation fixed the problem. Here's an example which works as expected:

```html
<div>
  <a href="/search/locations/d" data-navigo>one</a>
  <a href="/search/notlocations/d" data-navigo>two</a>
</div>
<hr />
<div id="content"></div>
<script src="/navigo.js"></script>
<script>
  window.addEventListener("load", () => {
    const router = new Navigo("/");
    const render = (content) =>
      (document.querySelector("#content").innerHTML = content);

    router.on({
      "/search/:v/:q": {
        as: "search",
        uses: function (match) {
          render("Search " + JSON.stringify(match.data));
        },
      },
      "/locations": {
        as: "home",
        uses: function () {
          render("Home");
        },
      },
    });
  });
</script>
```

P.S.
Version 8.0.0 is not officially released but can be installed via `npm install navigo@beta` or `yarn add navigo@beta`. Migration guide is available here https://github.com/krasimir/navigo/blob/big-rewrite/CHANGELOG.md#migration-guide and the new documentation here https://github.com/krasimir/navigo/blob/big-rewrite/DOCUMENTATION.md.
